### PR TITLE
[ENH] SQL: save user name and password via credentials manager

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
User name and password should not be saved in Settings.

##### Description of changes
Use credentials manager.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by astaric** [comment]
Not sure how this is supposed to work since username/password depend on the server/database.

To reproduce, create a workflow that connects to two databases using different credentials. Save it, reload it and see what happens.

**Comment by jerneju** [comment]
@astaric: Well, besides username and password also server name/address/other could be saved. And not just one, more of them.

File: Orange/widgets/data/owsql.py [hunk 1]
```diff
@@ -50,12 +51,15 @@ class OWSql(OWWidget):
     want_main_area = False
     resizing_enabled = False
 
+    cm_sql_username = CredentialManager("SQL username")
+    cm_sql_password = CredentialManager("SQL password")
```
Comment (kernc) [inline Orange/widgets/data/owsql.py]: Could this be, as per CredentialManager's docstring, a single `cm = CredentialManager('SQL Table')` line, further below used as `self.cm.username`, `self.cm.password`?

**Comment by kernc** [comment]
In that case, `cm` would need to be constructed per input values: `cm = CredentialManager('SQL - Table - {hostname}:{port}'.format(...))`. Credentials wouldn't propagate with saved workflows.

What's the issue with Settings again?

File: Orange/widgets/data/owsql.py (hunk 1 above)
Comment (jerneju) [inline Orange/widgets/data/owsql.py]: What about `"SQL Table: {}:{}".format(self.host, self.port)`

**Comment by jerneju** [comment]
@kernc 
> What's the issue with Settings again?
Error reporting sends workflow with that data.

**Comment by astaric** [comment]
Looks ok to me. Could you also add a migration that moves credentials from widget to Credentials Manager (it will be used when loading an old workflow).

Bonus points if you add some tests :)

File: Orange/widgets/data/owsql.py [hunk 2]
```diff
@@ -55,8 +55,8 @@ class Outputs:
     port = Setting(None)
     database = Setting(None)
     schema = Setting(None)
-    username = ""
-    password = ""
+    username = Setting(None)
```
Comment (astaric) [inline Orange/widgets/data/owsql.py]: What about adding a migrate_settings (class) method and storing the username and password to credentials manager there?

Comment (jerneju) [inline Orange/widgets/data/owsql.py]: Ok, I will do that.

Comment (jerneju) [inline Orange/widgets/data/owsql.py]: Fixed.

**Pull request merged**